tests/ExamplesTest.php

```diff
@@ -128,6 +128,23 @@ public static function provideExamples(): Generator
             'expectedOutput' => $expectedOutput,
         ];
 
+        /* Note: Do not assert output beyond the initial topology events, as it
+         * may vary based on the test environment. PHPUnit's matching behavior
+         * for "%A" also seems to differ slightly from run-tests.php, otherwise
+         * we could assert the final topologyClosed event. */
```

Comment:
I looked through [StringMatchesFormatDescription](https://github.com/sebastianbergmann/phpunit/blob/9.5.28/src/Framework/Constraint/String/StringMatchesFormatDescription.php) but couldn't figure out exactly why it differs from `run-tests.php`. In my local testing, PHPUnit had trouble matching expected output _after_ `%A`, in contrast to the attached [sdam_logger.phpt](https://github.com/mongodb/mongo-php-library/files/12283265/sdam_logger.txt) file (ignore the `txt` extension, which was needed to appease GitHub's validation).

Relevant format in the `phpt` test is:

```
--EXPECTF--
topologyOpening: %x was opened

topologyChanged: %x changed from Unknown to %s

%A

topologyChanged: %x changed from %s to %s

topologyClosed: %x was closed

===DONE===
```

Reply:
The PHPUnit `StringMatchesFormatDescription` splits line-by-line. So you cannot match several lines with a single `%A`. This works on my machine:

```php
        $expectedOutput = <<<'OUTPUT'
topologyOpening: %x was opened

topologyChanged: %x changed from Unknown to %s

%A
%A
%A
%A
%A
%A
%A
%A
%A
%A

topologyChanged: %x changed from %s to %s

topologyClosed: %x was closed


OUTPUT;
```

Reply:
> The PHPUnit StringMatchesFormatDescription splits line-by-line

Can you point me to where that happens? I know that [StringMatchesFormatDescription](https://github.com/sebastianbergmann/phpunit/blob/main/src/Framework/Constraint/String/StringMatchesFormatDescription.php) has some logic to convert `\r\n` (e.g. https://github.com/sebastianbergmann/phpunit/commit/05db035d21de4f2499ea20a349eb2be082543c2d), but it looks like it defers to `preg_match` to handle multi-line strings.

A series of `%A` patterns on their own line should be redundant, as that could otherwise be written with `%S` (see: [EXPECTF](https://qa.php.net/phpt_details.php#expectf_section]). And the regex conversion seemingly _does_ attempt to handle newlines for `%a` and `%A` based on the conversion logic in `regularExpressionForFormatDescription()`:

```
'%s' => '[^\r\n]+',
'%S' => '[^\r\n]*',
'%a' => '.+',
'%A' => '.*',
```

Asserting the number of newlines in the output is not portable, as that will vary based on the topology used to run the test. A 3-member replica set is going to log more lines than a 2-mongos sharded cluster or standalone server.